src/java.net.http/share/classes/jdk/internal/net/http/Http2Connection.java

```diff
@@ -796,14 +798,20 @@ void processFrame(Http2Frame frame) throws IOException {
                 return;
             }
 
-            // While pus frame is not null, the only acceptable frame on this
+            // While push frame is not null, the only acceptable frame on this
             // stream is a Continuation frame
-            if (pcs != null) {
+            if (pushContinuationState != null) {
                 if (frame instanceof ContinuationFrame cf) {
-                    handlePushContinuation(stream, cf);
+                    try {
+                        handlePushContinuation(stream, cf);
+                    } catch (UncheckedIOException e) {
+                        debug.log("Error handling Push Promise with Continuation: " + e.getMessage(), e);
+                        protocolError(ResetFrame.PROTOCOL_ERROR, e.getMessage());
+                    }
                 } else {
                     // TODO: Maybe say what kind of frame was received instead
-                    protocolError(ErrorFrame.PROTOCOL_ERROR, "Expected Continuation frame but received another type");
+                    pushContinuationState = null;
+                    protocolError(ErrorFrame.PROTOCOL_ERROR, "Expected a Continuation frame but received " + frame);
```

Comment:
In all other places in this method we have `return;` just after a call to `protocolError`, except in the two places where your changes added one. For consistency you should probably add this `return;` statement, even if it's not strictly needed. It would avoid having to have to analyze the whole structure of the nested `if - then - else` to figure out that it's actually not needed.